common/src/main/java/tachyon/security/authentication/AuthenticationProviderFactory.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.security.authentication;
+
+import javax.security.sasl.AuthenticationException;
+import javax.security.sasl.SaslClient;
+import javax.security.sasl.SaslServer;
+
+public class AuthenticationProviderFactory {
+  public enum AuthenticationMethod {
+    /**
+     * SIMPLE AuthenticationMethod stands for PLAIN mechanism SASL negotiation
+     * CUSTOM AuthenticationMethod stands for delegating the implementation of
+     * {@link SaslServer} and {@link SaslClient} to user to customize the SASL
+     * negotiation
+     */
+    SIMPLE("SIMPLE"),
```

Comment:
Is this `SIMPLE` related to `AuthenticationFactory.AuthType.SIMPLE` in #1162 ? If yes, we should probably still keep name consistent, otherwise, we should differentiate .